csharp-linq/LINQInnerJoin/LinqInnerJoin/EnrolmentHelper.cs

```diff
@@ -0,0 +1,169 @@
+
+            var enrollements = enrollments  //Enrolment data source                            
+                               .Join(   // First Join - Joining Student with Enrolment
+                                   students,
+                                   enrolment_lv1 => enrolment_lv1.StudentId,
+                                   student => student.Id,
+                                   (enrolment_lv1, student) => new { enrolment_lv1, student }   // result from first join
+                                )
+                               .Join(   // Second Join - Joining Course with the result of first join 
+                                   courses,
+                                   enrolment_lv2 => enrolment_lv2.enrolment_lv1.CourseId,
+                                   course => course.Id,
+                                   (enrolment_lv2, course) => new { enrolment_lv2, course }  // result fromn second join
+                                )
+                               .Select(e => new      // Selecting result from the selection of two joins
+                               {
+                                   e.enrolment_lv2.enrolment_lv1.Id,
+                                   StudentName = e.enrolment_lv2.student.Name,
+                                   CourseName = e.course.Name,
+                               });
+
+            return enrollements;
+        }
+
+        public static IEnumerable<dynamic> FilterEnrollments(IEnumerable<Enrolment> enrollments)
+        {
+            var students = Student.GetDummyStudents();
+            var courses = Course.GetDummyCourses();
+
+            var enrollements = enrollments  //Enrolment data source            
+                               .Join(       // First Join - Joining Student with Enrolment
+                                  students,
+                                  enrolment_lv1 => enrolment_lv1.StudentId,
+                                  student => student.Id,
+                                  (enrolment_lv1, student) => new { enrolment_lv1, student })   // result from first join
+                               .Join(  // Second Join - Joining Course with the result of first join
+                                  courses,
+                                  enrolment_lv2 => enrolment_lv2.enrolment_lv1.CourseId,
+                                  course => course.Id,
+                                  (enrolment_lv2, course) => new { enrolment_lv2, course })  // result fromn second join
+                               .Select(e => new      // Selecting result from the selection of two joins
+                                {
+                                  e.enrolment_lv2.enrolment_lv1.Id,
+                                  StudentName = e.enrolment_lv2.student.Name,
+                                  CourseId = e.course.Id,
+                                  CourseName = e.course.Name,
+                                })
```

Comment:
I think this duplicate of what we have in `GetEnrollments` methods. So maybe use the method directly instead.

Reply:
updated with extracted function